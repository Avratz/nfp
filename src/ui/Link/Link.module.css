.link {
  color: var(--neutral-3);
  text-decoration: none;
  font-family: var(--font-dmSans);
}

.link:hover,
.link:focus,
.link:active {
  color: var(--neutral-1);
  text-decoration: none;
}
