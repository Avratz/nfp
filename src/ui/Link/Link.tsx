import { cx } from 'class-variance-authority'
import { default as NextLink, LinkProps as NextLinkProps } from 'next/link'
import { forwardRef } from 'react'

import styles from './Link.module.css'

export interface LinkProps extends NextLinkProps {
  className?: string
  children: React.ReactNode
}

export const Link = forwardRef<HTMLAnchorElement, LinkProps>(
  function forwardLink({ className, children, ...props }, ref) {
    return (
      <NextLink className={cx(styles.link, className)} {...props} ref={ref}>
        {children}
      </NextLink>
    )
  },
)
