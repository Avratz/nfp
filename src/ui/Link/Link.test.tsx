import { render } from '@testing-library/react'
import { Link } from './Link'

describe('Link', () => {
  it('renders default', () => {
    const { container } = render(<Link href="href">Link</Link>)
    expect(container).toMatchSnapshot()
  })
})
