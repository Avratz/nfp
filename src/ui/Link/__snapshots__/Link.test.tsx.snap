// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link renders default 1`] = `
<div>
  <a
    class="link"
    href="href"
  >
    Link
  </a>
</div>
`;
